cli/command_test.go

```diff
@@ -381,6 +383,117 @@ func TestBaseCommand_Prompt_Cancels(t *testing.T) {
 	}
 }
 
+// Tests back-and-forth conversation using Prompt().
+func TestBaseCommand_Prompt_Dialog(t *testing.T) {
+	t.Parallel()
+
+	cmd := &RootCommand{}
+
+	stdinReader, stdinWriter := io.Pipe()
+	stdoutReader, stdoutWriter := io.Pipe()
+
+	cmd.SetStdin(stdinReader)
+	cmd.SetStdout(stdoutWriter)
+
+	ctx := context.Background()
+	errCh := make(chan error)
+	var color, iceCream string
+	go func() {
+		defer close(errCh)
+		var err error
+		color, err = cmd.Prompt(ctx, "Please enter a color: ")
+		if err != nil {
+			errCh <- err
+			return
+		}
+
+		iceCream, err = cmd.Prompt(ctx, "Please enter a flavor of ice cream: ")
+		if err != nil {
+			errCh <- err
+			return
+		}
+	}()
+
+	readWithTimeout(t, stdoutReader, "Please enter a color:")
+	writeWithTimeout(t, stdinWriter, "blue\n")
+	readWithTimeout(t, stdoutReader, "Please enter a flavor of ice cream:")
+	writeWithTimeout(t, stdinWriter, "mint chip\n")
+
+	select {
+	case err := <-errCh:
+		if err != nil {
+			t.Fatal(err)
+		}
+	case <-time.After(time.Second):
+		t.Fatal("timed out waiting for background Prompt() goroutine to exit")
+	}
+
+	if color != "blue" {
+		t.Fatalf(`got color %q, wanted "blue"`, color)
+	}
+	if iceCream != "mint chip" {
+		t.Fatalf(`got iceCream %q, wanted "mint chip"`, iceCream)
+	}
+}
+
+// readWithTimeout does a single read from the given reader. It calls Fatal if
+// that read fails or the returned string doesn't contain wantSubStr. May leak a
+// goroutine on timeout.
+func readWithTimeout(t *testing.T, r io.Reader, wantSubstr string) {
```

Comment:
Nit: prefer testing.TB interface instead (in case we ever needed to use this in benchmarks)